File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int l = 0 , su = 0 , r = height.size()-1;
        while (l<r){
            su = max(su ,min(height[l],height[r]) * (r-l));
            if (height[l] < height[r]) l ++;
            else r--;
        } 
        return su ;  
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left = 0
        right = len(height)-1
        maximum_water = float('-inf')
        while left<right:
            maximum_water = max(maximum_water , min(height[left] , height[right]) * (right - left))
            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        return maximum_water
            